src/include/txn.i

```diff
@@ -480,31 +485,59 @@ __txn_visible_id(WT_SESSION_IMPL *session, uint64_t id)
  * __wt_txn_visible --
  *	Can the current transaction see the given ID / timestamp?
  */
-static inline bool
+static inline int
 __wt_txn_visible(
-    WT_SESSION_IMPL *session, uint64_t id, const wt_timestamp_t *timestamp)
+    WT_SESSION_IMPL *session, uint64_t id, const wt_timestamp_t *timestamp,
+    WT_UPDATE *upd)
```

Comment:
It seems wrong to me that we now have `__wt_txn_visible` taking an update.  I expected any logic that required an update to stay in `__wt_txn_upd_visible` and this function just deals with the visibility of a transaction ID.